install_config/aggregate_logging_sizing.adoc

```diff
@@ -0,0 +1,255 @@
+...
+...
+
+----
+
+Notice how the pods are placed in different cluster nodes. +
+
+
+By default the amount of RAM allocated to each ES instance is 8GB. `*ES_INSTANCE_RAM*` is the parameter used in the openshift-ansible link:https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_examples/files/examples/v1.3/infrastructure-templates/enterprise/logging-deployer.yaml[template]. Keep in mind that *half* of this value will be passed to the individual elasticsearch pods java processes link:https://www.elastic.co/guide/en/elasticsearch/guide/current/heap-sizing.html#_give_half_your_memory_to_lucene[heap size].
+
+Further information regarding installing EFK can be found in the openshift-docs repository xref:../install_config/aggregate_logging.adoc[documentation]. 
+
+[[install-config-aggregate-logging-sizing-guidelines-rate-limiting]]
+== Systemd-journald and rsyslog 
+
+*Rate-limiting*
+
+In RHEL7 the `systemd-journald.socket` unit creates `/dev/log` during the boot process, and then passes input to systemd-journald.service. Every `syslog()` call goes to the journal. +
+Rsyslog uses the `imjournal` module as a default input mode for journal files. Refer to link:https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/7/html-single/System_Administrators_Guide/index.html#s1-interaction_of_rsyslog_and_journal[Interaction of rsyslog and journal] for detailed information about this topic.
+
+
+At Red Hat, we have used a simple test harness which uses link:http://linux.die.net/man/1/logger[logger] across the cluster nodes to make entries of different sizes at different rates in the system log. During testing simulations under a default RHEL7 installation at certain logging rates ( ~ 40 log lines per second ), we've hit the default rate limit of `rsyslogd`.
+
+*Scaling up* +
+
+As you scale up your project, the default logging environment might need some adjustments.
+After upgrading to `systemd-219-22.el7.x86_64`, we've added:
+
+----
+$IMUXSockRateLimitInterval 0
+$IMJournalRatelimitInterval 0
+----
+
+to `/etc/rsyslog.conf` and:
+
+----
+# Disable rate limiting
+RateLimitInterval=1s
+RateLimitBurst=10000
+Storage=volatile
+Compress=no
+MaxRetentionSec=5s
+----
+
+to `/etc/systemd/journald.conf`. 
+
+Now restart the services.
+
+----
+$ systemctl restart systemd-journald.service
+$ systemctl restart rsyslog.service
+----
+
+These settings accommodate for the 'bursty' nature of uploading in bulk. 
+After removing the rate limit, you may see increased CPU utilization on the system logging daemons as it processes any messages that would have previously been throttled.
+
+
+Rsyslog is configured (see ratelimit.interval, ratelimit.burst) to rate-limit entries read from the journal at 10,000 messages in 300 seconds. A good rule of thumb is to ensure that the rsyslog rate-limits always accommodate to the systemd-journald rate-limits.
```

Comment:
"ensure that the rsyslog rate-limits account for the systemd-journald rate-limits."